### PR TITLE
docs: add language specific guide for c++

### LINKED ISSUE #19898: Add language specific guide for C++
### Is this a docs issue?

- [X] My issue is about the documentation content or website

### Type of issue

Other

### Description

There is no language specfic guide for C++ unlike Go, Node.js, etc.

### Location

https://docs.docker.com/language

### Suggestion

I would like to fix this gap by adding a guide for that. If that sounds interesting, feel free to assign me this issue.

### PR DESCRIPTION
**Pull request by Pradumnasaraf**
## Description

- Add language specific guide for c++

## Related issues or tickets

fix #19898 

## Reviews

- [x] Technical review
- [ ] Editorial review
- [ ] Product review

### CODE REVIEW

**Comment by craig-osterhout** [comment]
Hi @Pradumnasaraf, 
Thank you!
We'll definitely work on getting this added.
I haven't had time to do a full review, but a few quick things:
- You can add the topics to the left navigation by updating the `data/toc.yaml` file
- It looks like there's a validation error in the `content/language/_index.md` file related to the link. Your folder name is `c++`, but the link is `c-plus-plus`.

**Comment by Pradumnasaraf** [comment]
Hey, @craig-osterhout.

Thank you for the feedback. I have updated the docs as you requested.

**Comment by Pradumnasaraf** [comment]
Hey,
Let me know if you need any help with the PR.

File: content/language/c-plus-plus/containerize.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+## Overview
+
+This section walks you through containerizing and running a C++ application.
+
+## Get the sample application
+
+Clone the sample application to use with this guide. Open a terminal, change directory to a directory that you want to work in, and run the following command to clone the repository:
+
+```console
+$ git clone https://github.com/Pradumnasaraf/c-plus-plus-docker.git
```
Comment (craig-osterhout) [inline content/language/c-plus-plus/containerize.md:23]: Can we fork your repo to a Docker-controlled repo and update this?

**Review by craig-osterhout — CHANGES_REQUESTED** [review]
Thanks again @Pradumnasaraf for your contribution. I left some review comments.
Also, feel free to add an acknowledgement callout for yourself like https://github.com/docker/docs/blob/main/content/language/golang/_index.md?plain=1#L11

File: content/language/c-plus-plus/containerize.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+---
+title: Containerize a C++ application
+keywords: C++, containerize, initialize
+description: Learn how to containerize a C++ application.
+aliases:
+  - /language/c-plus-plus/build-images/
```
[the remark this answers is not in the document]
Comment (Pradumnasaraf) [reply content/language/c-plus-plus/containerize.md]: Good catch, Craig 😅

File: content/language/c-plus-plus/containerize.md (hunk 1 above)
Comment (Pradumnasaraf) [reply content/language/c-plus-plus/containerize.md:20]: Yes, go ahead, Craig.

**Comment by Pradumnasaraf** [comment]
Hey @craig-osterhout, thanks for the review. I have addressed all of them.
Also, I have added an acknowledgement callout, thanks for that.

**Comment by dvdksn** [comment]
Hello. I just pushed a small change to use a different, shorter subpath for this guide: from `/c-plus-plus` to `/cpp`

**Comment by Pradumnasaraf** [comment]
Hey,
I hope I have made all the required changes. Let me know if I missed anything or if you need any help with the PR.

File: content/language/cpp/develop.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+services:
+  ok-api:
+    image: ok-api
+    build:
+      context: .
+      dockerfile: Dockerfile
+    ports:
+      - "8080:8080"
+    stdin_open: true
+    tty: true
```
Comment (dvdksn) [inline content/language/cpp/develop.md:43]: nit: remove these properties
```suggestion
```

**Review by dvdksn — COMMENTED** [review]
@craig-osterhout did you already start moving this repo to `dockersamples`?

**Comment by craig-osterhout** [comment]
> @craig-osterhout did you already start moving this repo to `dockersamples`?

I just reached out to devrel to see if they can help fork it. I don't have permissions for the dockersamples org. I'll update here when I hear back.

File: content/language/cpp/containerize.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+## Overview
+
+This section walks you through containerizing and running a C++ application.
+
+## Get the sample application
+
+Clone the sample application to use with this guide. Open a terminal, change directory to a directory that you want to work in, and run the following command to clone the repository:
+
+```console
+$ git clone https://github.com/Pradumnasaraf/c-plus-plus-docker.git
```
Comment (craig-osterhout) [inline content/language/cpp/containerize.md:20]:
```suggestion
$ git clone https://github.com/dockersamples/c-plus-plus-docker.git
```

**Comment by Pradumnasaraf** [comment]
Updated the link, @craig-osterhout. Thank you!

**Review by craig-osterhout — APPROVED** [review]
🚀
Thank you.

**Comment by Pradumnasaraf** [comment]
Thank you for the merge, Craig. Did it take some time to go live?

**Comment by dvdksn** [comment]
@Pradumnasaraf I kicked the publishing workflow now, the guide should be up on the website in a few minutes! 🎉 

Thank you for contributing this guide!

**Pull request merged**